### PR TITLE
Use AndroidX, Gradle 8.2, AGP 8.2.2, protobuf 3.25.2 and several nitpicky things

### PR DESCRIPTION
**Pull request by weebl2000**
No clue if this is wanted, but my Android Studio was warning me a lot and this seems to work.

### CODE REVIEW

**Comment by roussosalex** [comment]
Thanks for contributing (again)! Could you please rebase this PR on the updated `v2` branch?

**Comment by weebl2000** [comment]
> Thanks for contributing (again)! Could you please rebase this PR on the updated `v2` branch?

Sure, I've reapplied the suggestions on the latest v2 branch.

And nitpicked a little bit on the build files to make them a little more consistent.

Did not test all functionality extensively but it seems to build and works on my Pixel 6 running QPR 14 / 3.1(AP11.231215.009)

Using AndroidX packages removes the need for the lint-baseline.xml - so I've gone ahead and removed that.

File: app/src/main/assets/html/desfire-info.en.html [hunk 1]
```diff
@@ -31,4 +31,4 @@ <h5>Technical details</h5>
 thread. After that has happened, the Android watchdog thread will resume operation and detect the
 removal of the card.</p>
 </body>
-</html>
\ No newline at end of file
+</html>
```
Comment (roussosalex) [inline app/src/main/assets/html/desfire-info.en.html]: Fixing newlines and quotes should be in a separate PR

**Review by roussosalex — CHANGES_REQUESTED** [review]
Much needed update to AndroidX. Thanks!

File: app/src/main/assets/html/desfire-info.en.html (hunk 1 above)
Comment (weebl2000) [reply app/src/main/assets/html/desfire-info.en.html]: Done, removed any files that only have newlines added and reverted all lines that only had quotation marks changed.

**Review by roussosalex — APPROVED** [review]

**Review by kleest — APPROVED** [review]

**Pull request merged**